test/lib/jdk/test/lib/security/DataFetcher.java

```diff
@@ -0,0 +1,112 @@
+    ///
+    /// Users can also specify the "jdk.tests.repos.pattern" system property
+    /// to fetch the extracted file in an alternative location. The value of
+    /// this system property is a [format][java.util.Formatter] string that
+    /// takes the last part of [organization name][Artifact#organization()],
+    /// the [name][Artifact#name()], the [version][Artifact#revision()],
+    /// and `entry` as arguments. For example:
+    ///
+    /// With the [CMS_ML_DSA] class inside this test, the pattern
+    /// `file:///Users/tester/repos/external/%s/%s/%4$s` will be resolved to
+    /// a local file like `/Users/tester/repos/external/lamps-wg/cms-ml-dsa/entry`;
+    /// and a pattern `https://raw.repos.com/%s/%s/%s/%s` will be resolved to
+    /// a URL like `https://raw.repos.com/lamps-wg/cms-ml-dsa/c8f0cf7/entry`.
+    ///
+    /// @param klass the `Artifact` class
+    /// @param zipPrefix the common prefix for each entry in the ZIP file
+    /// @param entry the entry name without `zipPrefix`
+    public static byte[] fetchData(Class<?> klass, String zipPrefix, String entry)
+            throws IOException {
+        Artifact artifact = klass.getAnnotation(Artifact.class);
+        var org = artifact.organization();
+        var prop = System.getProperty("jdk.tests.repos.pattern");
+        if (prop != null && org.startsWith("jpg.tests.jdk.repos.")) {
+            var url = String.format(prop,
+                    org.substring(org.lastIndexOf('.') + 1),
+                    artifact.name(),
+                    artifact.revision(),
+                    entry);
+            System.out.println("fetching " + url + "...");
+            try (var is = new URI(url).toURL().openStream()) {
+                return is.readAllBytes();
+            } catch (URISyntaxException e) {
+                throw new IOException("Invalid URL: " + url);
+            }
+        } else {
+            try {
+                Path p = ArtifactResolver.resolve(klass).entrySet().stream()
+                        .findAny().get().getValue();
+                System.out.println("fetching " + zipPrefix + entry + " in " + p + "...");
+                try (ZipFile zf = new ZipFile(p.toFile())) {
+                    ZipEntry ze = zf.getEntry(zipPrefix + entry);
+                    if (ze != null) {
+                        return zf.getInputStream(ze).readAllBytes();
+                    } else {
+                        throw new RuntimeException("Entry not found: " + entry);
+                    }
+                }
+            } catch (ArtifactResolverException e) {
+                throw new SkippedException("Cannot find the artifact " + artifact.name(), e);
+            }
+        }
+    }
+
+    @Artifact(
+            organization = "jpg.tests.jdk.repos.lamps-wg",
+            name = "dilithium-certificates",
+            revision = "785a549",
+            extension = "zip",
+            unpack = false)
+    public static class DILITHIUM_CERTIFICATES {
```

Comment:
I think these should be defined in the tests where they are used.

Reply:
Soon, `DILITHIUM_CERTIFICATES` will be used by KAT for RFC 9881 and whatever [draft-ietf-lamps-kyber-certificates](https://datatracker.ietf.org/doc/draft-ietf-lamps-kyber-certificates/) will be. We can revise the structure later.